### ISSUE TITLE
Cureent way to add friends causes issue

### ISSUE THREAD

**Issue by danrobi11** [irislib/iris-electron#15]
Send friend request instead of both side exchanging profile links. 

Better scenario:

1- Bob post his Iris profile link on his website.

2- Alice copy Bob profile from Bob website

3- Now Alice can follow Bob. (Alice will now receive feeds from Bob, but they cant private message yet. Bob doesnt know Alice at this point)

4- Now Alice send friend request to Bob. 

5- Bob receive a friend request from Alice.

6- Bob accepts Alice friend request. (Therefore, by accepting Alice friend request, Bob is now following Alice)

7- Now Alice and Bob can send private messages to eachother and Bob will now receive feeds from Alice.

**Comment by mmalmi** [comment]
You can send your chat invite link to someone, and you'll discover each other through it, exchanging only one link.

In a decentralized system, allowing friend requests from anyone would be vulnerable to spam.

**Comment by danrobi11** [comment]
> You can send your chat invite link to someone, and you'll discover each other through it, exchanging only one link.

No. Tested and that doesnt work. Both party need to exchange profile link.  The friend wasnt able to see me at all until I've send him my profile link.
Therefore, its impossible to friend someone you dont know. 

> In a decentralized system, allowing friend requests from anyone would be vulnerable to spam.

I disagree. Thats why the block option is there. To block trolls.
